src/main/java/com/forgeessentials/api/permissions/WorldZone.java

```diff
@@ -51,6 +51,29 @@ public void afterLoad()
             zone.worldZone = this;
     }
     
+    public AreaZone getAreaZone(int id)
+    {
+        for (AreaZone zone : getAreaZones())
+        {
+            if (zone.getId() == id)
+            {
+                return zone;
+            }
+        }
+        return null;
+    }
+
+    public int nextAreaZoneId()
+    {
+        int desiredID = getServerZone().getMaxZoneID() + 1;
+        while (desiredID < 5 || getServerZone().getAreaZone(desiredID) != null)
```

Comment:
What is the reasoning for this logic?  Specifically the desiredID < 5.

Reply:
Sorry, I should've commented that part. In practice, getAreaZone() doesn't return values when dealing within the server zone( id 1 I believe), or the world zone (id 2 I believe). So, this forces it to proceed until at least id of 5 to account for those, and to leave space just incase other such zones are added later.

Reply:
Ok.  Going forward, I would recommend not using hardcoded values for constants.  For example, `desiredID < 5` should be replaced with `desiredID < RESERVED_ID_COUNT`, where reserved_id_count is defined as `private static final int RESERVED_ID_COUNT = 5;`.  For this specific line, the risk is low and the logic is sound so I don't think you need to replace this value with a constant, but as a best practice, it is good to use constants in place of hardcoded values.